zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/Leader.java

```diff
@@ -173,6 +174,13 @@ public BufferStats getProposalStats() {
     void addForwardingFollower(LearnerHandler lh) {
         synchronized (forwardingFollowers) {
             forwardingFollowers.add(lh);
+            /*
+            * Any changes on forwardiongFollowers could possible affect the need of Oracle.
+            * */
+            QuorumVerifier currentQv = self.getQuorumVerifier();
+            if (currentQv instanceof QuorumOracleMaj) {
```

Comment:
i think we can make the code nicer and eliminate instanceof checks like this one by adding updateNeedOracle to the QuorumVerifier interface and have empty implementations in the other implementations of QuorumVerifier.

Reply:
oh, i see we already do that! i think we can skip the instance checks then. this can all boil down to self.getQuorumVerifier().updateNeedOracle(forwardingFollowers); do we need the new ArrayList<>(..)?